### PR TITLE
chore: cleanup MomentsController

### PR DESCRIPTION
**Pull request by salzig**
# Description 

#840 states that the Controller should be refactored. I tried to clean it up a bit, without changing too much. 

# Test Coverage
✅

### CODE REVIEW

File: app/controllers/moments_controller.rb [hunk 1]
```diff
@@ -7,23 +7,20 @@ class MomentsController < ApplicationController
...
-      @user_logged_in = true
-      period = 'day'
       # +1 day buffer to ensure we include today as well
       end_date = Date.current + 1.day
-      start_date = get_start_by_period(period, end_date)
-      @react_moments = Moment.where(user: current_user)
-                             .group_by_period(period,
+      start_date = end_date - 1.week
+      @react_moments = current_user.moments
+                             .group_by_period('day',
```
Comment (julianguyen) [inline app/controllers/moments_controller.rb]: Rubocop is complaining that `.group_by_period` needs to be aligned with `.moments` on line 20.

File: app/controllers/moments_controller.rb [hunk 2]
```diff
@@ -103,46 +91,33 @@ def moment_params
...
     @category = Category.new
-    @moods = Mood.where(user: current_user).order(created_at: :desc)
+    @moods = current_user.moods.order(created_at: :desc)
     @mood = Mood.new
     @strategies = associated_strategies
     @strategy = Strategy.new
   end
 
+  def load_viewers
+    @viewers = current_user.allies_by_status(:accepted)
```
Comment (julianguyen) [inline app/controllers/moments_controller.rb]: Nice!

File: app/controllers/moments_controller.rb [hunk 3]
```diff
@@ -103,46 +91,33 @@ def moment_params
...
+
   def associated_strategies
     # current_user's strategies and all viewable strategies from allies
-    strategy_ids = current_user.strategies.pluck(:id)
-    @viewers.each do |ally|
-      ally.strategies.each do |strategy|
-        strategy_ids << strategy.id if strategy.viewer?(current_user)
-      end
+    strategy_ids = current_user.strategy_ids
+    Strategy.where(user: @viewers).uniq.each do |strategy|
```
Comment (julianguyen) [inline app/controllers/moments_controller.rb]: Is uniq necessary?

**Comment by salzig** [comment]
🎉

**Pull request merged**